@copilot+claude-sonnet-4.6 - this change is a backport from `master` but fails in the test code due to other intermediate changes on `master`. These are the errors from the maven build. Adjust the test case accordingly:
```
11:59:02.779  [INFO] [****-event-spy] Generated /home/****/workspace/knime-core_PR-91@tmp/withMaven359a97b1/maven-spy-20260410-115712-4815689684892933823799.log
11:59:02.779  [ERROR] Failed to execute goal org.eclipse.tycho:tycho-compiler-plugin:4.0.7:compile (default-compile) on project org.knime.core.tests: Compilation failure: Compilation failure: 
11:59:02.779  [ERROR] /home/****/workspace/knime-core_PR-91/org.knime.core.tests/src/org/knime/core/node/logging/KNIMELoggerTest.java:[342] 
11:59:02.779  [ERROR] 	assertNull(StartupLogRouter.resolveFailsafeTarget(null, EclipseUtil.Application.WARMSTART),
11:59:02.779  [ERROR] 	                                                                                ^^^^^^^^^
11:59:02.779  [ERROR] WARMSTART cannot be resolved or is not a field
11:59:02.779  [ERROR] /home/****/workspace/knime-core_PR-91/org.knime.core.tests/src/org/knime/core/node/logging/KNIMELoggerTest.java:[353] 
11:59:02.779  [ERROR] 	assertNull(StartupLogRouter.resolveFailsafeTarget("", EclipseUtil.Application.WARMSTART),
11:59:02.779  [ERROR] 	                                                                              ^^^^^^^^^
11:59:02.779  [ERROR] WARMSTART cannot be resolved or is not a field
11:59:02.779  [ERROR] 2 problems (2 errors)
```

--- a/org.knime.core.tests/src/org/knime/core/node/logging/KNIMELoggerTest.java
+++ b/org.knime.core.tests/src/org/knime/core/node/logging/KNIMELoggerTest.java
@@ -339,8 +339,8 @@ class KNIMELoggerTest {
         assertEquals(System.out, StartupLogRouter.resolveFailsafeTarget(null, EclipseUtil.Application.EXECUTOR),
             "Expected System.out by default for EXECUTOR");
         // disabled by default for all other applications
-        assertNull(StartupLogRouter.resolveFailsafeTarget(null, EclipseUtil.Application.WARMSTART),
-            "Expected null (disabled) by default for WARMSTART");
+        assertNull(StartupLogRouter.resolveFailsafeTarget(null, EclipseUtil.Application.UNKNOWN),
+            "Expected null (disabled) by default for UNKNOWN");
     }
 
     @Test
@@ -350,8 +350,8 @@ class KNIMELoggerTest {
             "Expected System.out for blank value with AP");
         assertEquals(System.out, StartupLogRouter.resolveFailsafeTarget("  ", EclipseUtil.Application.EXECUTOR),
             "Expected System.out for whitespace-only value with EXECUTOR");
-        assertNull(StartupLogRouter.resolveFailsafeTarget("", EclipseUtil.Application.WARMSTART),
-            "Expected null (disabled) for blank value with WARMSTART");
+        assertNull(StartupLogRouter.resolveFailsafeTarget("", EclipseUtil.Application.UNKNOWN),
+            "Expected null (disabled) for blank value with UNKNOWN");
     }
 
     @Test
@@ -362,7 +362,7 @@ class KNIMELoggerTest {
         // message is still in the buffer since the disabled failsafe did not drain it
         final var wasDrained = new AtomicBoolean();
         router.drainToLogger(m -> wasDrained.set(true));
-        assertTrue(wasDrained.get(), "Expected message to remain buffered after disabled failsafe");
+        assertTrue(wasDrained.get(), "Expected message to stay buffered until explicitly drained to the logger");
     }
 
     @Test

